### PR TITLE
Campaign implementation

### PR DESCRIPTION
**Pull request by linkielink**
This PR introduces campaigns for assets, based on the settings inside the `chainConfig`.

An asset can now be flagged as a `campaignAsset` and will show additional info on the UI.
There are two types of campaigns right now; `apy` and `points_with_multiplier`.
The idea is to show additional external protocol rewards based on the campaigns parameter.

This implementation covers:
- underlying auto-compounding Stride APYs
- droplets earned by deposits and farms

Missing implementation due to third party is the updated rewards center, as there is no API to query accumulated Droplets from.

### CODE REVIEW

File: src/components/Modals/Farm/FarmModalContentHeader.tsx [hunk 1]
```diff
@@ -46,38 +50,77 @@ export default function FarmModalContentHeader(props: Props) {
...
+        .dividedBy(2)
+        .shiftedBy(-PRICE_ORACLE_DECIMALS)
+    }
+
+    const astroLpPosition = updatedAccount?.stakedAstroLps.find(
+      (position) => position.denom === farm.denoms.lp,
+    )
+    if (!astroLpPosition) return BN_ZERO
+
+    return astroLpPosition.amount.dividedBy(2)
```
Comment (StefChatz) [inline src/components/Modals/Farm/FarmModalContentHeader.tsx:68]: Personal preference but maybe we can shorten this to: 

`return astroLpPosition?.amount.dividedBy(2) || BN_ZERO`

**Review by StefChatz — CHANGES_REQUESTED** [review]

**Comment by Patricie29** [comment]
lgtm

File: src/components/common/assets/AssetCampaignCopy.tsx [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+  asset: Asset
+  size: 'xs' | 'sm' | 'base' | 'lg' | 'xl'
+  className?: string
+  textClassName?: string
+  withLogo?: boolean
+  amount?: BigNumber
+}
+
+export default function AssetCampaignCopy(props: Props) {
+  const { asset, className, amount, withLogo, size, textClassName } = props
```
Comment (bobthebuidlr) [inline src/components/common/assets/AssetCampaignCopy.tsx:20]: Could be nice to split this up in two components. A Controller where you check for `asset.campaign` and the actual component. Will save some unnecessary renders as well of the hooks that you are now rendering regardless

Comment (bobthebuidlr) [reply src/components/common/assets/AssetCampaignCopy.tsx:20]: Or you could also create a type extensions of Asset, like `AssetWithCampaign`. Because you already know that the asset that is passed to this component has the campaign.

File: src/components/Modals/Farm/FarmBorrowings.tsx [hunk 3]
```diff
@@ -186,15 +186,16 @@ export default function FarmBorrowings(props: FarmBorrowingsProps) {
...
+
         if (!asset || !maxAmount)
           return <React.Fragment key={`input-${coin.denom}`}></React.Fragment>
         return (
           <TokenInput
             key={`input-${coin.denom}`}
             amount={coin.amount}
             asset={asset}
-            max={maxAmount.plus(coin.amount)}
+            max={maxAmount.isNaN() ? BN_ZERO : maxAmount.plus(coin.amount)}
```
Comment (bobthebuidlr) [inline src/components/Modals/Farm/FarmBorrowings.tsx:198]: This check seems unnecessary to me. It cannot be NaN. In line 189 you either set amount (BN) or BN_ZERO.

**Review by bobthebuidlr — CHANGES_REQUESTED** [review]

File: src/hooks/campaign/useCampaignAprs.tsx [hunk 4]
```diff
@@ -0,0 +1,30 @@
...
+      if (campaignApis.length === 0) return [] as StakingApr[]
+
+      const campaignAprs = Promise.all(
+        [...new Set(campaignApis)].map((api) => getStakingAprs(api ?? '')) ?? [],
+      )
+      return (await campaignAprs).flat()
+    },
+    {
+      suspense: true,
+      fallback: [] as StakingApr[],
```
[the remark this answers is not in the document]
Comment (linkielink) [reply src/hooks/campaign/useCampaignAprs.tsx:27]: For now, it is. This needs an overhaul if there are other APIs than Stride's in the future. But currently, this way of typing is sufficient. 
This is just a base layer for future campaigns, so I went the route that added the least amount of code.

File: src/hooks/assets/useAssetsNoOraclePrices.ts [hunk 5]
```diff
@@ -25,8 +28,8 @@ export default function useAssetsNoOraclePrices() {
     async () => fetchSortAndMapAllAssets(chainConfig, assetParams, fetchedPerpsParams),
     */
 
-    assetParams && `chains/${chainConfig.id}/noOraclePrices`,
-    async () => fetchSortAndMapAllAssets(chainConfig, assetParams),
+    assetParams && campaignAprs && `chains/${chainConfig.id}/noOraclePrices`,
```
[the remark this answers is not in the document]
Comment (linkielink) [reply src/hooks/assets/useAssetsNoOraclePrices.ts:31]: It's a useless check. You're right. Checking for `campaignsAprs.length` when there are `campaignAssets` is not going to work, as `points_with_multiplier` campaigns have no APRs.

**Review by StefChatz — APPROVED** [review]
